framework/auth/core.py

```diff
@@ -859,6 +862,11 @@ def confirm_email(self, token, merge=False):
 
         self.update_search_nodes()
 
+        # log to sentry if somehow the username isn't in the email list
+        # This is for data gathering purposes, so we can find when this is happening
+        if self.username.lower() not in self.emails:
```

Comment:
This doesn't seem necessary, since you're already ensuring that this will be `False` ~L848.